psydac/linalg/solvers.py

```diff
@@ -101,7 +104,7 @@ class ConjugateGradient(InverseLinearOperator):
     [1] A. Maister, Numerik linearer Gleichungssysteme, Springer ed. 2015.
 
     """
-    def __init__(self, A, *, x0=None, tol=1e-6, maxiter=1000, verbose=False):
+    def __init__(self, A, *, x0=None, x0update='static', tol=1e-6, maxiter=1000, verbose=False):
```

Comment:
After the discussion at the weekly developers' meeting, it appears that we can achieve the same functionality without this flag. In the `__init__(self, A, *, x0=None, **kwargs)`:
```python
assert isinstance(A, LinearOperator)
assert A.domain.dimension == A.codomain.dimension
domain = A.codomain
codomain = A.domain

if x0 is not None:
    assert isinstance(x0, Vector)
    assert x0.space is codomain
else:
    x0 = codomain.zeros()

self._A = A
self._domain = domain
self._codomain = codomain
self._x0 = x0
```

And in `dot(self, b, *, out=None)`:
```python
assert isinstance(b, Vector)
assert b.space is self.domain

if out is not None:
    assert isinstance(out, Vector)
    assert out.space is self.codomain

# TO DO: Make sure that y.copy(out=y) returns immediately y
x = self._x0.copy(out=out)
```

Reply:
e45247f022b2ee118316c84d8c7c5f72167c9fdb